bin/wasm-node/javascript/src/client.ts

```diff
@@ -599,58 +410,55 @@ export function start(options?: ClientOptions): Client {
         }
       }
 
-      // Build a promise that will be resolved or rejected after the chain has been added.
-      // TODO: because of https://github.com/microsoft/TypeScript/issues/11498 we need to define the callbacks as possibly null, and go through `unknown`
-      let chainAddedPromiseResolve;
-      let chainAddedPromiseReject;
-      const chainAddedPromise: Promise<Chain> = new Promise((resolve, reject) => {
-        chainAddedPromiseResolve = resolve;
-        chainAddedPromiseReject = reject;
-      });
-
-      pendingConfirmations.push({
-        ty: 'chainAdded',
-        reject: chainAddedPromiseReject as unknown as (error: AddChainError) => void,
-        resolve: chainAddedPromiseResolve as unknown as (c: Chain) => void,
-        jsonRpcCallback: options.jsonRpcCallback,
-      });
-
-      postMessage(worker, {
-        ty: 'addChain',
-        chainSpec: options.chainSpec,
-        databaseContent: typeof options.databaseContent === 'string' ? options.databaseContent : "",
-        potentialRelayChains: potentialRelayChainsIds,
-        jsonRpcRunning: !!options.jsonRpcCallback,
-      });
-
-      return chainAddedPromise;
+      const outcome = await instance.addChain(options.chainSpec, typeof options.databaseContent === 'string' ? options.databaseContent : "", potentialRelayChainsIds, options.jsonRpcCallback);
+
+      if (!outcome.success)
+        throw new AddChainError(outcome.error);
+
+      const chainId = outcome.chainId;
+      const wasDestroyed = { destroyed: false };
+
+      // `expected` was pushed by the `addChain` method.
+      // Resolve the promise that `addChain` returned to the user.
+      const newChain: Chain = {
+        sendJsonRpc: (request) => {
+          if (alreadyDestroyedError)
+            throw alreadyDestroyedError;
+          if (wasDestroyed.destroyed)
+            throw new AlreadyDestroyedError();
+          if (!options.jsonRpcCallback)
+            throw new JsonRpcDisabledError();
+          if (request.length >= 8 * 1024 * 1024)
+            return;
+          instance.request(request, chainId);
+        },
+        databaseContent: (maxUtf8BytesSize) => {
+          if (alreadyDestroyedError)
+            return Promise.reject(alreadyDestroyedError);
+          if (wasDestroyed.destroyed)
+            throw new AlreadyDestroyedError();
+          return instance.databaseContent(chainId, maxUtf8BytesSize);
+        },
+        remove: () => {
+          if (alreadyDestroyedError)
+            throw alreadyDestroyedError;
+          if (wasDestroyed.destroyed)
+            throw new AlreadyDestroyedError();
+          wasDestroyed.destroyed = true;
+          console.assert(chainIds.has(newChain));
```

Comment:
is this assert needed?

Reply:
Asserts are never "needed". It's here to detect bugs and provide better error messages. I'm not sure what you're suggesting, but if you're suggesting that it you should be removed, then I'd say no, because it's here to detect a potential bug.

Reply:
Im not suggesting removal - just wondering if this is there intentionally or erroneously.